common/extension/metadata_remote.go

```diff
@@ -0,0 +1,25 @@
+package extension
+
+import (
+	"fmt"
+	"github.com/apache/dubbo-go/metadata/remote"
+	perrors "github.com/pkg/errors"
+)
+
+var (
+	creator func() (remote.RemoteMetadataService, error)
```

Comment:
another question: are you sure the creator do not need a lock to protect read/write?